scripts/tasks/src/swc.ts

```diff
@@ -1,27 +1,59 @@
-import { exec } from 'child_process';
-import { promisify } from 'util';
+import fs from 'fs';
+import path from 'path';
 
+import { transform } from '@swc/core';
 import type { Options as SwcOptions } from '@swc/core';
-import { logger } from 'just-scripts';
-
-const execAsync = promisify(exec);
+import glob from 'glob';
+import * as match from 'micromatch';
 
 type Options = SwcOptions & { module: { type: 'es6' | 'commonjs' | 'amd' } };
 
-function swcCli(options: Options) {
+async function swcTransform(options: Options) {
   const { outputPath, module } = options;
-  const swcCliBin = 'npx swc';
-  const sourceDirMap = {
-    es6: 'src',
-    commonjs: 'lib',
-    amd: 'lib',
-  };
-  const sourceDir = sourceDirMap[options.module.type];
-
-  const cmd = `${swcCliBin} ${sourceDir} --out-dir ${outputPath} --config module.type=${module?.type}`;
-  logger.info(`Running swc CLI: ${cmd}`);
-
-  return execAsync(cmd);
+  let sourceFiles: string[] = [];
+
+  if (module.type === 'es6') {
+    sourceFiles = glob.sync('src/**/*.{ts,tsx}');
```

Comment:
to mitigate duplication we can probably extract root folder name to separate constant:

```diff
const sourceRootDirName = module.type === 'es6' ? 'src' : 'lib'

// then we can do

if (module.type === 'es6') {
-    sourceFiles = glob.sync('src/**/*.{ts,tsx}');
+    sourceFiles = glob.sync(`${sourceRootDirName}/**/*.{ts,tsx}`);
}

// later in code

const compiledFilePath = path.resolve(
      process.cwd(),
-      fileName.replace(`${module.type === 'es6' ? 'src' : 'lib'}`, outputPath!),
+      fileName.replace(sourceRootDirName, outputPath!),     
    );

```